### PR TITLE
Feature/update log file rotation

### PR DESCRIPTION
**Pull request by schroda**

### CODE REVIEW

File: AndroidCompat/Config/src/main/java/xyz/nulldev/ts/config/Logging.kt [hunk 1]
```diff
@@ -66,16 +71,44 @@ private fun getLogger(name: String): ch.qos.logback.classic.Logger {
...
+    maxFileSize: String,
+    maxTotalSize: String,
+) {
+    val logger = getBaseLogger()
+
+    val appender = logger.getAppender(FILE_APPENDER_NAME) as RollingFileAppender<*>? ?: return
+    val rollingPolicy = appender.rollingPolicy as SizeAndTimeBasedRollingPolicy<*>
+    rollingPolicy.apply {
+        maxHistory = maxFiles
+        setMaxFileSize(FileSize.valueOf(maxFileSize))
```
Comment (Syer10) [inline AndroidCompat/Config/src/main/java/xyz/nulldev/ts/config/Logging.kt:112]: How forgiving is this? I don't want to cause a server crash if a client sends a invalid value like `10gbb`

Comment (schroda) [reply AndroidCompat/Config/src/main/java/xyz/nulldev/ts/config/Logging.kt:112]: it did not crash when sending the mutation, however, once an invalid value was in the config file the server crashed on startup

Comment (Syer10) [reply AndroidCompat/Config/src/main/java/xyz/nulldev/ts/config/Logging.kt:112]: Maybe make a extension function that wraps FileSize.valueOf in a try/catch with a default if its invalid, I want to protect against manual file editing errors as well

Comment (schroda) [reply AndroidCompat/Config/src/main/java/xyz/nulldev/ts/config/Logging.kt:112]: because it's a java class a companion extension function can't be added

**Review by Syer10 — APPROVED** [review]

**Comment by Syer10** [comment]
@schroda Needs to be formatted

**Pull request merged**